library/src/main/java/com/google/android/exoplayer2/extractor/flv/ScriptTagPayloadReader.java

```diff
@@ -79,18 +86,33 @@ protected void parsePayload(ParsableByteArray data, long timeUs) throws ParserEx
       return;
     }
     int type = readAmfType(data);
-    if (type != AMF_TYPE_ECMA_ARRAY) {
-      // Should never happen.
-      throw new ParserException();
-    }
-    // Set the duration to the value contained in the metadata, if present.
-    Map<String, Object> metadata = readAmfEcmaArray(data);
-    if (metadata.containsKey(KEY_DURATION)) {
-      double durationSeconds = (double) metadata.get(KEY_DURATION);
-      if (durationSeconds > 0.0) {
-        durationUs = (long) (durationSeconds * C.MICROS_PER_SECOND);
+//    if (type != AMF_TYPE_ECMA_ARRAY) {
+//      //   Should never happen.
+//      //aom: actually it can happen for live streams
+//      throw new ParserException();
+//
+//    }
+    if (type == AMF_TYPE_ECMA_ARRAY) {
+      // Set the duration to the value contained in the metadata, if present.
+      Map<String, Object> metadata = readAmfEcmaArray(data);
+      if (metadata.containsKey(KEY_DURATION)) {
+        double durationSeconds = (double) metadata.get(KEY_DURATION);
+        if (durationSeconds > 0.0) {
+          durationUs = (long) (durationSeconds * C.MICROS_PER_SECOND);
+        }
+      }
+      if (metadata.containsKey(KEY_FRAMES)) {
+        HashMap<String, List<Double>> keyFrames = (HashMap<String, List<Double>>) metadata.get(KEY_FRAMES);
+        setKeyFrameTimes(keyFrames.get(KEY_FRAME_TIMES));
```

Comment:
It might be nice to convert the values into long microsecond and byte offset arrays here, rather than keeping them as doubles. That way the public getter methods, for time at least, are all in the same unit (i.e. microseconds).